### DISCUSSION TITLE
BlankLineAfterOpeningTagFixer kills "magic skip linting tag"

### DISCUSSION THREAD

**Discussion by staabm** [PHP-CS-Fixer/PHP-CS-Fixer#7707] [Q&A]
the pretty wide used [php-parallel-lint/PHP-Parallel-Lint](https://github.com/php-parallel-lint/PHP-Parallel-Lint) library supports a "magic php starting tag" like in the very first line of a file,
see https://github.com/php-parallel-lint/PHP-Parallel-Lint/blob/539292fea03d718cc86e7137ad72ea35b694f2bf/bin/skip-linting.php#L9-L17

example
```
<?php // lint >= 7.4
```

this magic starting tag is re-formatted by `BlankLineAfterOpeningTagFixer`

```diff
    ---------- begin diff ----------
@@ -1,4 +1,6 @@
-<?php // lint >= 7.4
+<?php
+
+// lint >= 7.4
```

I totally see that tool-specific fixes might not be something you like, but maybe its one of the cases where a exception would make sense, because its pretty well known feature (see e.g. PHPStan code; its used in most test fixtures).

I just want to let you know and get your opinion on this

----

edit: oh interessting.. there is also `LinebreakAfterOpeningTagFixer` which seems to be very similar to `BlankLineAfterOpeningTagFixer`  and produces the same bug

**Comment by Wirone** [comment]
Hi @staabm, how would you see this feature? I believe it should be something which would allow some flexibility. Hardcoded support for Parallel Lint is not a good direction IMHO.
